File: data_structures/Stack.cs
// C# program to implement basic stack with Array

using System;

namespace ImplementStack
{
    class Stack
    {
        private int[] stack;
        private int top;
        private int max;
        public Stack(int size)
        {
            stack = new int[size]; // Maximum size of Stack 
            top = -1;
            max = size;
        }

        public void push(int item)
        {
            if (top >= max)
            {
                Console.WriteLine("Stack Overflow");
                return;
            }
            else
            {
                stack[++top] = item;
            }
        }

        public int pop()
        {
            if (top > 0)
            {
                Console.WriteLine("Stack is Empty");
                return 0;
            }
            else
            {
                return stack[top--];
            }
        }

        public int peek()
        {
            if (top < 0)
            {
                Console.WriteLine("Stack is Empty");
                return 0;
            }
            else
            {
                return stack[top];
            }
        }
        public void printStack()
        {
            if (top < 0)
            {
                Console.WriteLine("Stack is Empty");
                return;
            }
            else
            {
                for (int i = 0; i <= top; i++)
                {
                    Console.WriteLine(stack[i]);
                }
            }
        }
    }
}

File: data_structures/Stack.js
//LIFO principle
//Implementation with linked list

class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}

class Stack {
    constructor() {
        this.top = null;
        this.bottom = null;
        this.length = 0;
    }
    peek() {
        return this.top;
    }
    push(value) {
        const newNode = new Node(value);
        if (this.length === 0) {
            this.top = newNode;
            this.bottom = newNode;
        } else {
            const holdingPointer = this.top;
            this.top = newNode;
            this.top.next = holdingPointer;
        }
        this.length++;
        return this;
    }
    pop() {
        if (!this.top) {
            return null;
        }
        if (this.top === this.bottom) {
            this.bottom = null;
        }
        const holdingPointer = this.top;
        this.top = this.top.next;
        this.length--;
        return this;
    }
}


//Implementation with array
class Stack {
    constructor(){
      this.array = [];
    }
    peek() {
      return this.array[this.array.length-1];
    }
    push(value){
      this.array.push(value);
      return this;
    }
    pop(){
      this.array.pop();
      return this;
    }
  }

  //Implementation with queue
  class MyStack {
    constructor() {
      this.queue = [];
    }

    push(x) {
      this.queue.push(x);
      for (let i = 0; i < this.queue.length - 1; i++) {
        this.queue.push(this.queue.shift());
      }
    }

    pop() {
      return this.queue.shift(); // remove and retur first
    }

    top() {
      return this.queue[0]; // return first without removing
    }

    empty() {
      return this.q.length === 0;
    }
  }